### PR TITLE
Drain fix

### PR DESCRIPTION
**Pull request by metalgearsloth**
There was a return in the loop. Also made some other optimisations at the same time.

### CODE REVIEW

File: Content.Server/Fluids/EntitySystems/DrainSystem.cs [hunk 1]
```diff
@@ -24,27 +30,33 @@ public override void Update(float frameTime)
                 }
                 drain.Accumulator -= drain.DrainFrequency;
 
-                /// Best to do this one every second rather than once every tick...
-                _solutionSystem.TryGetSolution(drain.Owner, DrainComponent.SolutionName, out var drainSolution);
+                if (!managerQuery.TryGetComponent(drain.Owner, out var manager))
+                    continue;
```
Comment (mirrorcult) [inline Content.Server/Fluids/EntitySystems/DrainSystem.cs:34]: just put it in the `EntityQuery`?

File: Content.Server/Fluids/EntitySystems/DrainSystem.cs [hunk 2]
```diff
@@ -24,27 +30,33 @@ public override void Update(float frameTime)
...
                 _solutionSystem.SplitSolution(drain.Owner, drainSolution, (drain.UnitsDestroyedPerSecond * drain.DrainFrequency));
 
-                /// This will ensure that UnitsPerSecond is per second...
+                // This will ensure that UnitsPerSecond is per second...
                 var amount = drain.UnitsPerSecond * drain.DrainFrequency;
-                var xform = Transform(drain.Owner);
-                List<PuddleComponent> puddles = new();
+
+                if (!xformQuery.TryGetComponent(drain.Owner, out var xform))
+                    continue;
```
Comment (mirrorcult) [inline Content.Server/Fluids/EntitySystems/DrainSystem.cs:49]: same here

File: Content.Server/Fluids/EntitySystems/DrainSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/Fluids/EntitySystems/DrainSystem.cs:34]: Seemed better considering 90% of the time the float is just getting accumulated and we don't need the solutioncontainermanager

Comment (mirrorcult) [reply Content.Server/Fluids/EntitySystems/DrainSystem.cs:34]: i suppose so

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**